File: Semestru 4/MPP/lab/.net/lab3/lab3/Services/services/IServiceSignup.cs
using Model.app.domain;

namespace Services.services
{
    public interface IServiceSignup
    {
        IEnumerable<Signup> GetAll();
        Signup? GetById(Tuple<int, int> id);
        Signup? Add(Child child, Event event_);
        Signup? Update(Signup signup);
        IEnumerable<Signup>? GetByEvent(int eventId);
        IEnumerable<Signup>? GetByChild(int childId);
    }
}

File: Semestru 4/MPP/lab/java/lab3/Services/src/main/java/org/example/services/IServiceSignup.java
package org.example.services;

import org.example.domain.Child;
import org.example.domain.Event;
import org.example.domain.Signup;
import org.example.domain.Tuple;

public interface IServiceSignup {
    Iterable<Signup> GetAll();
    Signup AddSignup (Child child, Event event);
    Signup GetById (int childId, int eventId);
    Iterable<Signup> GetByEvent (int eventId);
    Iterable<Signup> GetByChild (int childId);
    Signup Delete (int childId, int eventId);
    Iterable<Signup> GetAllMapped(IServiceChild serviceChild, IServiceEvent serviceEvent);
}
